desc/objectives/_geometry.py

```diff
@@ -1019,3 +1019,148 @@ def compute(self, params, constants=None):
             profiles=constants["profiles"],
         )
         return data["L_grad(B)"]
+
+
+class GoodCoordinates(_Objective):
+    """Target "good" coordinates, meaning non self-intersecting curves.
+
+    Uses a method by Z. Tecchiolli et al, minimizing
+
+    1/ρ² |√g|² + ω |𝐞ᵨ|²
+
+    where √g is the jacobian of the coordinate system and 𝐞ᵨ is the covariant radial
+    basis vector.
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Equilibrium that will be optimized to satisfy the Objective.
+    omega : float
```

Comment:
Is this in their paper/presentation? or could we change this name? as `omega` is already the name of the (currently unused) toroidal stream function, so this could get confusing. Although if we want to stick to their convention then it is probably fine

Reply:
it was used in slides he showed at spectaculars but i don't think its that important, we can change it